tools/release/release-output/output-validations.ts

```diff
@@ -27,6 +30,41 @@ export function checkReleaseBundle(bundlePath: string): string[] {
   return failures;
 }
 
+/**
+ * Checks the specified TypeScript definition file by ensuring it does not contain invalid
+ * dynamic import statements. There can be invalid dynamic imports paths because we compose the
+ * release package by moving things in a desired output structure. See Angular package format
+ * specification and https://github.com/angular/material2/pull/12876
+ */
+export function checkTypeDefinitionFile(filePath: string): string[] {
+  const baseDir = dirname(filePath);
+  const fileContent = readFileSync(filePath, 'utf8');
+  const failures = [];
+
+  let match: RegExpMatchArray;
+
+  // Walk through each dynamic import and ensure that the import path is valid within the release
+  // output. Note that we don't want to enforce that there are no dynamic imports because type
+  // inference is heavily used within the schematics and is useful in some situations.
+  while ((match = dynamicImportStatement.exec(fileContent)) !== null) {
```

Comment:
Wouldn't this be more reliable and less error prone if we did it with the AST? It shouldn't add much to the build time since we don't have to create a project and we don't have to interact with the type checker.

Reply:
So, I looked into using the TypeScript AST, even though there is some overhead and it feels slower, it's probably just nicer than a Regex (as said above). We also should be able to use the parsed AST for additional/future checks for these files.